src/zenml/zen_server/routers/steps_endpoints.py

```diff
@@ -244,26 +261,111 @@ def get_step_status(
 
 @router.get(
     "/{step_id}" + LOGS,
-    responses={401: error_response, 404: error_response, 422: error_response},
+    responses={
+        400: error_response,
+        401: error_response,
+        404: error_response,
+        422: error_response,
+    },
 )
 @async_fastapi_endpoint_wrapper
 def get_step_logs(
     step_id: UUID,
-    offset: int = 0,
-    length: int = 1024 * 1024 * 16,  # Default to 16MiB of data
-    strip_timestamp: bool = False,
+    count: int = DEFAULT_PAGE_SIZE,
+    level: int = LoggingLevels.INFO.value,
+    search: Optional[str] = None,
+    page: Optional[int] = None,
+    from_file_index: Optional[int] = None,
+    from_position: Optional[int] = None,
     _: AuthContext = Security(authorize),
-) -> str:
-    """Get the logs of a specific step.
+) -> List[LogEntry]:
+    """Get log entries for efficient pagination.
+
+    This endpoint supports optimized navigation modes:
+
+    - page=1: Returns the latest entries (first page)
```

Comment:
First page = latest entries? I thought first page are the first entries?

Reply:
It is. Since I changed the approach, I forgot to adjust the docstring.